File: farm_wise_frontend/lib/models/order.dart
class Order {
  String orderId,
      dealerId,
      productId,
      paymentMode,
      productName,
      dealerName,
      dealerMobile,
      quantityUnit;
  int basePrice, dealPrice, orderQtyLots, orderValue, minNoLot, quantityPerLot;
  bool isOrderAccepted,
      isClosed,
      halfPaymentDone,
      fullPaymentDone,
      isSellerClosed,
      isBuyerClosed;
  DateTime? closedOn,
      requiredOnOrBefore,
      expectingResponseBefore,
      recievedOn,
      sellerClosedOn,
      buyerClosedOn;
  List productImages;

  Order({
    required this.dealerId,
    required this.orderId,
    required this.productId,
    required this.minNoLot,
    required this.quantityPerLot,
    required this.productName,
    required this.paymentMode,
    required this.halfPaymentDone,
    required this.fullPaymentDone,
    required this.basePrice,
    required this.dealPrice,
    required this.orderQtyLots,
    required this.orderValue,
    required this.isOrderAccepted,
    required this.isClosed,
    required this.closedOn,
    required this.requiredOnOrBefore,
    required this.expectingResponseBefore,
    required this.recievedOn,
    required this.productImages,
    required this.dealerMobile,
    required this.dealerName,
    required this.quantityUnit,
    required this.buyerClosedOn,
    required this.isBuyerClosed,
    required this.isSellerClosed,
    required this.sellerClosedOn,
  });
}

File: farm_wise_backend/model/order.js
const mongoose = require("mongoose");

const OrderSchema = new mongoose.Schema(
  {
    sellerId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
    },
    buyerId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
    },
    productId: {
      type: mongoose.Schema.Types.ObjectId,
      required: true,
    },
    basePrice: {
      type: Number,
      required: true,
    },
    dealPrice: {
      type: Number,
      required: true,
    },
    orderQtyLots: {
      type: Number,
      required: true,
    },
    orderValue: {
      type: Number,
      required: true,
    },
    isOrderAccepted: {
      type: Boolean,
      required: true,
    },
    paymentMode: {
      type: String,
      required: true,
    },
    halfPaymentDone: {
      type: Boolean,
      required: true,
    },
    fullPaymentDone: {
      type: Boolean,
      required: true,
    },
    isClosed: {
      type: Boolean,
    },
    isSellerClosed: {
      type: Boolean,
    },
    isBuyerClosed: {
      type: Boolean,
    },
    sellerClosedOn: {
      type: Date,
    },
    buyerClosedOn: {
      type: Date,
    },
    closedOn: {
      type: Date,
    },
    requiredOnOrBefore: {
      type: Date,
    },
    expectingResponseBefore: {
      type: Date,
    },
  },
  {
    timestamps: true,
  }
);

const Order = mongoose.model("t_orders", OrderSchema);

module.exports = Order;
